airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -122,6 +125,44 @@ def ti_update_state(
         )
     elif isinstance(ti_patch_payload, TITerminalStatePayload):
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+    elif isinstance(ti_patch_payload, TIDeferredStatePayload):
+        trigger_row = Trigger(
+            classpath=ti_patch_payload.classpath,
+            kwargs=ti_patch_payload.kwargs,
+            created_date=ti_patch_payload.created_date,
+        )
+        session.add(trigger_row)
+        session.flush()
+
+        ti = session.query(TI).filter(TI.id == ti_id_str).one_or_none()
```

Comment:
We already have this in `old` variable:

https://github.com/apache/airflow/blob/4d3140dba5976acb38c3dffd4fd963514f8da634/airflow/api_fastapi/execution_api/routes/task_instances.py#L78

Reply:
Not needed anymore because we added the TODO for execution timeout. Removing this code